rasa/core/policies/policy.py

```diff
@@ -428,6 +435,21 @@ def __eq__(self, other: Any) -> bool:
         if not isinstance(other, PolicyPrediction):
             return False
 
+        # The `diagnostic_data` values can be numpy arrays, so we have to check
```

Comment:
I don't think we should consider diagnostic data in checking if Prediction is the same. Where is it used?

Reply:
I agree. The `__eq__` implementation is mostly used for testing, but I think considering the diagnostic data is still quite unintuitive.